### ISSUE TITLE
improve review summary

### ISSUE THREAD

**Issue by wbazant** [falling-fruit/falling-fruit-web#446]
I noticed my test review with just a picture is dragging down an overall score, unlike on the live site:
[
![Screenshot from 2024-09-02 11-07-38](https://github.com/user-attachments/assets/b8df4f20-1dd8-4642-8581-f204e379d2c5)
](url)

![Screenshot from 2024-09-02 11-09-10](https://github.com/user-attachments/assets/2413ce62-9581-4752-8b4d-8b32303d7e09)
d

The proposed resolution is to skip empty scores, so that an average of 3, empty, and 5 is 4, not 3.

**Comment by wbazant** [comment]
Relatedly, #416 proposes:
- Move the review summary to the location overview (or remove)

I'm thinking that maybe the review summary should go away entirely, instead of getting fixed. The new layout allows for plenty of space for reviews, I think locations typically don't have many reviews - maybe 1 or 2, there's no need to summarize them - and it's questionable whether taking an average is the right way of summarizing reviews.

Also, relatedly:  'does this location have reviews, and how many' is in itself a summary, and a very objective one! https://github.com/falling-fruit/falling-fruit-web/issues/364#issuecomment-2336664843 proposes to only show overview/reviews tab if there are reviews.

**Comment by wbazant** [comment]
I agree the review summary is less good over the reviews, and some summary being present in the overview could be good, especially in the half-open drawer: there's currently a needless surprise in the UX when the overview | reviews top ribbon pops up.

Thanks for mentioning [1517357](`https://www.fallingfruit.org/locations/1517357`), it is a good example of reviews working well!

Comparing it to https://beta.fallingfruit.org/locations/1517357, I also like the quality scores being done with words instead of with stars, and they would help to make the review summary shorter. HTML mockup:
![Screenshot from 2024-10-04 09-16-27](https://github.com/user-attachments/assets/1742f152-ea59-41e5-b2a0-78f6938aef66)


 @ezwelty what do you think about the stars for scores? I think they could remain as an input feature if hovering on stars makes the corresponding word appear on the screen, like:
![Screenshot from 2024-10-04 08-49-47](https://github.com/user-attachments/assets/bb60148f-313c-43ef-aae4-e3147afd0ef6)
Then the reviews list would either use words and stars or just words, and the overview would use just words.


So I would propose changes as follows:
- on mobile, move the review summary into the overview tab
- fix the algorithm (remove empty scores when summarising)
- change the review form and review list to use qualitative scores for 1-5 values
- change the UI of overview summary to only use words, be visually coherent, and fit on one line on mobile
- keep the two-column UI on desktop, just check it looks okay (something wrong with the whitespace there currently)

**Comment by ezwelty** [comment]
@wbazant In your mockup, I would probably drop the number of reviews and show number of reviews in parentheses after each label-value pair. I would also suggest formatting the labels and values differently, as on the old website, for legibility.

**Quality** Good 3.2/5 (1) · **Yield** Excellent 5.0/5 (2)
**Unripe fruit** August (1), September (1) · **Ripe fruit** September (1)

Regarding how to input and display the ratings... We started with words (Poor, Fair, Good, etc) but these are loaded terms that don't always translate well (between people or languages). It is also hard to numerically average them. So gradually we started to treat them as numbers.

I'm not sure what the best way is, but tempted to suggest using stars/numbers only:

- [ ] Make the review form stars require a click. The current hover is buggy and makes it look like a rating has been set even when it hasn't.
- [ ] In review, display either as 5 stars with selected stars orange (`* * * x x`), or as a number and an orange star (`3 *`). Either way, I think the stars should immediately follow their associated label (**Quality** or **Yield**) and layout should allow both be on the same line.
- [ ] In overview, show the numeric average and either 5 stars with (partially) selected stars orange or an orange star. Again layout should allow both ratings to sit on the same line.
